mempool/clist_mempool.go

```diff
@@ -176,6 +176,15 @@ func (mem *CListMempool) Unlock() {
 	mem.updateMtx.Unlock()
 }
 
+// Safe for concurrent use by multiple goroutines.
+func (mem *CListMempool) PreUpdate() {
+	rechecking := !mem.recheck.done()
```

Comment:
```suggestion
	if !mem.config.Recheck { return }
	
	rechecking := !mem.recheck.done()
```

Reply:
We don't need to consider the config. If it's disabled, the pre-existing code handles that case correctly